disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -37,12 +37,12 @@ import           Prelude
 -- | Return a vector whose size is within the provided bounds
 vectorOfSize :: Int -> Int -> Gen a -> Gen [a]
 vectorOfSize min' max' gen =
-  chooseSize min' max' >>= flip vectorOf gen
+  choose (min', max') >>= flip vectorOf gen
 
--- | Return an 'Int' which is between the provided range, and influenced by the current "size"
+-- | Return an 'Int' which is between the provided range, and influenced by the current "size" -- deprecated, please use "choose" instead
 chooseSize :: Int -> Int -> Gen Int
 chooseSize min' max' =
-  sized (return . min max' . max min')
+  choose (min', max')
```

Comment:
Given the description I would have expected the implementation to be something like:
```hs
chooseSize :: Int -> Int -> Gen Int
chooseSize x0 y0 =
  sized $ \sz0 ->
    let
      x =
        toInteger x0

      y =
        toInteger y0

      sz =
        toInteger sz0

      diff =
        ((y - x) * sz) `quot` 99
    in
      choose (x0, fromIntegral $ x + diff)
```

Reply:
Actually that probably needs something to clamp `x + diff` so that it doesn't go larger than `y0`